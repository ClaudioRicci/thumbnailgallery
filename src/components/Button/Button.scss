.button {
  border-radius: 1em;
  background: #ffa500;
  color: #000;
  text-align: center;
  font-size: 1.4em;
  padding: 0.5em 2vw;
  width: 70%;
  transition: all 0.5s;
  margin: 3vh 0 0;
}

.button span {
  position: relative;
}

.button span:after {
  content: "\00bb";
  position: absolute;
  opacity: 0;
  top: 0;
  right: -20px;
}

.button span,
.button span:after {
  transition: 0.5s;
}

.button,
.button span {
  cursor: pointer;
  display: inline-block;
}

.button:hover span {
  padding-right: 1em;
}

.button:hover span:after {
  opacity: 1;
  right: 0;
}

.button:hover {
  background-color: #62c462;
  color: #fff;
}

.closeButton,
.getOfferButton {
  position: absolute;
  right: 2%;
  width: auto;
  z-index: 1;
}

.closeButton {
  top: -50px;
}

.getOfferButton {
  bottom: 16px;
}
