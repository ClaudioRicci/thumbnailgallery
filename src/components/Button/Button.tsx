import React from "react";
import "./Button.scss";
import { pure } from "recompose";
import { ButtonProps } from "../../interfaces/interfaces";

const Button: React.SFC<ButtonProps> = props => {
  return (
    <button
      data-testid="button"
      className={props.buttonType}
      title={props.title}
    >
      <span>{props.label}</span>
    </button>
  );
};

Button.defaultProps = {
  buttonType: "button",
  title: "Launch Gallery",
  label: "Launch Gallery"
};

export default pure(Button);
