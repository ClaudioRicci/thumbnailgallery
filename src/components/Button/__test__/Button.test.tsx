import React from "react";
import ReactDOM from "react-dom";
import Button from "../Button";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button></Button>, div);
});

test("Renders Button Correctly", done => {
  const { getByTestId } = render(<Button label="Close" />);
  expect(getByTestId("button")).toHaveTextContent("Close");
  done();
});

it("Matches snapshot", () => {
  const tree = renderer.create(<Button label="Close"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
