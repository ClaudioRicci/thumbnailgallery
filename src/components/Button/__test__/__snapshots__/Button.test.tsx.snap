// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<button
  className="button"
  data-testid="button"
>
  <span>
    Close
  </span>
</button>
`;
